lectures/zksnark.tex

```diff
@@ -0,0 +1,812 @@
+\documentclass[../lecture-notes.tex]{subfiles}
+
+\begin{document}
+
+\subsection{What the zk-SNARK is?}
+
+Let's first discuss what zk-SNARK is.
+
+\begin{definition}
+    \textbf{zk-SNARK} – Zero-Knowledge Succinct Non-interactive ARgument of Knowledge.
+\end{definition}
+
+But what do terms like ``argument of knowledge``, ``succinct``, ``non-interactive``, and 
```

Comment:
I guess after the "Introduction to Zero-Knowledge" lecture, we might take a more formal view of these terms (I might myself add a couple of lines here). Nonetheless, we still need to maintain simplistic explanations of what that is for better general understanding!